### PR TITLE
feat(core): add amplifyUuid and amplifyUrl with builtin polyfill loader for RN

### PR DESCRIPTION
**Pull request by HuiSF**
#### Description of changes

- Added `amplifyUrl` and `amplifyUuid` to replace `URL` (global) and `uuid` (package) to simplify the RN polyfill loader usage
- Refactored packages which used the above two dependencies

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/core/src/utils/amplifyUrl/index.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import './polyfill';
+
+const AmplifyUrl = URL;
+const AmplifyUrlSearchParams = URLSearchParams;
```
Comment (cshfang) [inline packages/core/src/utils/amplifyUrl/index.ts:7]: Do we know of any implementation differences that we might want to at least call out in comments? That `pathname` thing comes to mind

**Review by cshfang — APPROVED** [review]
Yaaas

File: packages/analytics/src/providers/personalize/utils/cachedSession.ts [hunk 2]
```diff
@@ -2,8 +2,7 @@
 // SPDX-License-Identifier: Apache-2.0
 
 import { Cache } from '@aws-amplify/core';
-import { isBrowser } from '@aws-amplify/core/internals/utils';
-import { v4 as uuid } from 'uuid';
+import { isBrowser, amplifyUuid } from '@aws-amplify/core/internals/utils';
```
[the remark this answers is not in the document]
Comment (HuiSF) [reply packages/analytics/src/providers/personalize/utils/cachedSession.ts:5]: So far only the `v4` is used in the codebase, if something else need, we can add and refactor accordingly.

File: packages/core/src/utils/amplifyUrl/index.ts (hunk 1 above)
Comment (HuiSF) [reply packages/core/src/utils/amplifyUrl/index.ts:7]: Cool I'll add a comment in the PR refactoring `TextEncoder`

**Review by AllanZhengYP — APPROVED** [review]

**Pull request merged**